src/core/creative.js

```diff
@@ -0,0 +1,130 @@
+import { SortedListItem } from '../util/sorted-list-item'
+import { UniversalAdId } from './universal-ad-id'
+
+/**
+ * A creative.
+ *
+ * @class Creative
+ * @extends SortedListItem
+ * @author Tim De Pauw <tim.depauw@zentrick.com>
+ * @copyright © 2018 Zentrick nv
+ */
+export class Creative extends SortedListItem {
+  constructor () {
+    super()
+    this._universalAdId = new UniversalAdId()
```

Comment:
It feels to me like we want to set `._linear`, `._nonLinearAds` and `._companionAds` to `null` in the constructor. Right now, they're `undefined` by default which feels to me like something is wrong.

However, that opens a can of worms, because it means setting every field of every class to an explicit default value.

Additionally, for `nonLinearAds` specifically, we're dealing with an array, I'm still not sure whether it should be nullable, because that forces people to first check if it's null and then check if it's empty. I'd rather initialize all array-typed values to an empty array and remove the setters.

And finally, it's currently hard to find out what type of creative you're dealing with, since you have to check `linear`, `nonLinearAds` _and_ `companionAds`. Perhaps it would be easier if `Creative` were a simple array of instances of `Linear`, `NonLinearAds` and `CompanionAds` so that you could do your own filtering.

Reply:
I am now using a single array to records the creative ads.